test/UsersController.test.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+
+import type { TokenObject } from '../src/types';
+
+import test from 'ava';
+import request from 'supertest-as-promised';
+import ouathClients from '../src/oauthClients.json';
+
+import { app } from '../src/main';
+import settings from '../src/settings';
+
+const USER_CREDENTIALS = {
+  password: 'password',
+  username: 'newUser@test.com',
+};
+
+let user;
+let userToken;
+
+test.serial('should return a new user object', async t => {
+  const response = await request(app)
+    .post('/v1/users')
+    .send(USER_CREDENTIALS);
+
+  user = response.body;
+
+  t.is(response.status, 200);
+  t.truthy(user.username === USER_CREDENTIALS.username);
+  t.truthy(user.id && user.passwordHash && user.salt && user.created_at);
+});
+
+test.serial('should throw an error if username is already exist', async t => {
+  const response = await request(app)
+    .post('/v1/users')
+    .send(USER_CREDENTIALS);
+
+  t.is(response.status, 400);
+  t.is(response.body.message, 'user with the username is already exist');
+});
+
+test.serial('should login the user', async t => {
+  const response = await request(app)
+    .post('/oauth/token')
```

Comment:
this endpoint doesn't belong to UsersController, its node-ouauth server. but I think this is okey place for test it.